wp-includes/js/service-worker-navigation-routing.js

```diff
@@ -31,29 +31,32 @@
 						return response;
 					}
 				}
-				const channel = new BroadcastChannel( 'wordpress-server-errors' );
-
-				// Wait for client to request the error message.
-				channel.onmessage = ( event ) => {
-					if ( event.data && event.data.clientUrl && url === event.data.clientUrl ) {
-						response.text().then( ( text ) => {
-							channel.postMessage({
-								requestUrl: url,
-								bodyText: text,
-								status: response.status,
-								statusText: response.statusText
-							});
-							channel.close();
-						} );
-					}
-				};
 
-				// Close the channel if client did not request the message within 30 seconds.
-				setTimeout( () => {
-					channel.close();
-				}, 30 * 1000 );
+				if ( canStreamResponse() ) {
+					return caches.match( ERROR_500_BODY_FRAGMENT_URL );
+				}
+
+				return response.text().then( function( errorText ) {
+					return caches.match( ERROR_500_URL ).then( function( errorResponse ) {
 
-				return caches.match( canStreamResponse() ? ERROR_500_BODY_FRAGMENT_URL : ERROR_500_URL );
+						if ( ! errorResponse ) {
+							return response;
+						}
+
+						return errorResponse.text().then( function( text ) {
+							let init = {
+								status: errorResponse.status,
+								statusText: errorResponse.statusText,
+								headers: errorResponse.headers
+							};
+
+							let body = text.replace( /<!--WP_SERVICE_WORKER_ERROR_MESSAGE-->/, errorMessages.error );
+							body = body.replace( /<!--WP_SERVICE_WORKER_ERROR_DETAILS-->/, errorText );
```

Comment:
@westonruter Should we have a placeholder text here in case there are no details for the error? For example `No details received for the error` or something similar?

Reply:
That's a good question. I don't think so, since if the `details` element is used then it would be expanded to find no additional details. The problem seems to be that the container element needs to be omitted if there is no error message, right?

Reply:
Maybe, I had the thought that for consistency it would always be good to display the `details` even if there are no details -- to state explicitly there are no details so that it would be clear there were no details. Otherwise perhaps the user wouldn't know that no details were found.

If you think it's better to just not display the `details` at all then we can do that instead. Thoughts?

Reply:
Maybe better for now to just omit the details entirely if there is no error message detected.

But I see that this presumes that the `details` isn't already present in the document.